tests/unit/datastore_test.py

```diff
@@ -90,6 +331,586 @@ def test_profile_update_rules(self):
+        """
+        def mock_read(path):
+            self.assertEqual(path, TEST_PROFILE_PATH)
+            return Mock(spec=EtcdResult)
+
+        self.etcd_client.read.side_effect = mock_read
+        self.assertTrue(self.datastore.profile_exists("TEST"))
+
+    def test_profile_exists_false(self):
+        """
+        Test profile_exists() when it doesn't exist.
+        """
+        def mock_read(path):
+            self.assertEqual(path, TEST_PROFILE_PATH)
+            raise KeyError()
+
+        self.etcd_client.read.side_effect = mock_read
+        self.assertFalse(self.datastore.profile_exists("TEST"))
+
+    def test_create_profile(self):
+        """
+        Test create_profile()
+        """
+        self.datastore.create_profile("TEST")
+        rules = Rules(id="TEST",
+                      inbound_rules=[Rule(action="allow",
+                                          src_tag="TEST"),
+                                     Rule(action="deny")],
+                      outbound_rules=[Rule(action="allow")])
+        expected_calls = [call(TEST_PROFILE_PATH + "tags", '["TEST"]'),
+                          call(TEST_PROFILE_PATH + "rules", rules.to_json())]
+        self.etcd_client.write.assert_has_calls(expected_calls, any_order=True)
+
+    def test_delete_profile(self):
+        """
+        Test deleting a policy profile.
+        """
+        self.datastore.delete_profile("TEST")
+        self.etcd_client.delete.assert_called_once_with(TEST_PROFILE_PATH,
+                                                        recursive=True,
+                                                        dir=True)
+
+    def test_get_profile_names_2(self):
+        """
+        Test get_profile_names() when there are two profiles.
+        """
+        self.etcd_client.read.side_effect = mock_read_2_profiles
+        profiles = self.datastore.get_profile_names()
+        self.assertSetEqual(profiles, {"UNIT", "TEST"})
+
+    def test_get_profile_names_no_key(self):
+        """
+        Test get_profile_names() when the key hasn't been set up.  Should
+        return empty set and not raise a KeyError.
+        """
+        self.etcd_client.read.side_effect = mock_read_profiles_key_error
+        profiles = self.datastore.get_profile_names()
+        self.assertSetEqual(profiles, set())
+
+    def test_get_profile_names_no_profiles(self):
+        """
+        Test get_profile_names() when there are no profiles.
+        """
+        self.etcd_client.read.side_effect = mock_read_no_profiles
+        profiles = self.datastore.get_profile_names()
+        self.assertSetEqual(profiles, set())
+
+    def test_get_profile_members(self):
+        """
+        Test get_profile_members() when there are endpoints.
+        """
+        self.etcd_client.read.side_effect = mock_read_4_endpoints
+        members = self.datastore.get_profile_members("TEST")
+        self.assertSetEqual({"567890abcdef", "7890abcdef12"},
+                            set(members))
```

Comment:
Why are you casting the members result to a set? That could be a destructive operation (e.g. in the case members = [1,2,2]). If you're trying to guarantee a stable ordering, you could use `members.sorted()`.